sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/rest_client.hpp

```diff
@@ -220,6 +376,19 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
          */
         Nullable<std::string> ContinuationToken;
       };
+    } // namespace _detail
+    /**
+     * @brief Response type for #Azure::Storage::Files::DataLake::PathClient::Undelete.
+     */
+    struct UndeletePathResult final
+    {
+      /**
+       * The type of the resource.  The value may be "file" or "directory".  If not set, the value
+       * is "file".
+       */
+      std::string ResourceType;
```

Comment:
IIRC, `undelete` was pulled from SDK. Can you check .Net SDK?

Reply:
In .Net SDK, They implemented the undeleted API.
On convenience layer, they return a path client of deletedPath.
```
        public virtual Response<DataLakePathClient> UndeletePath(
            string deletedPath,
            string deletionId,
            CancellationToken cancellationToken = default)
            => UndeletePathInternal(
                deletedPath,
                deletionId,
                async: false,
                cancellationToken)
                .EnsureCompleted();
```
On protocol layer, they just return a string instead of enum.
```
namespace Azure.Storage.Files.DataLake
{
    internal partial class PathUndeleteHeaders
    {
        private readonly Response _response;
        public PathUndeleteHeaders(Response response)
        {
            _response = response;
        }
        /// <summary> The type of the resource.  The value may be &quot;file&quot; or &quot;directory&quot;.  If not set, the value is &quot;file&quot;. </summary>
        public string ResourceType => _response.Headers.TryGetValue("x-ms-resource-type", out string value) ? value : null;
        /// <summary> Indicates the version of the Blob service used to execute the request. This header is returned for requests made against version 2009-09-19 and above. </summary>
        public string Version => _response.Headers.TryGetValue("x-ms-version", out string value) ? value : null;
    }
}
```

Reply:
Move this into `_detail` namespace.